src/core/workload.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Base objects for workload operations across VM + K8s charms."""
+from abc import ABC, abstractmethod
+
+from literals import PATHS
+
+
+class ZKPaths:
```

Comment:
I don't think this should live here Probably better to have `PathsBase()` and then declare `ZKPaths` under `src/workload.py`? You can check the example on the [kafka PR](https://github.com/canonical/kafka-operator/pull/161/files#diff-4d1ed63ca48f694e77680b1434284f1d8d896cffe1842fb6b8b8e616c192b696R10-R17)